### PR TITLE
feat(python): expose `convert_to_deltalake`

### LINKED ISSUE #1767: Expose convert_to_delta to Python API
# Description
Expose the new convert to delta functionality to python users. Can be done once this PR is merged: https://github.com/delta-io/delta-rs/pull/1686

### PR DESCRIPTION
**Pull request by ion-elgreco**
# Description
Exposes added `convert to delta` functionality by @junjunjd to Python API.  

# Related Issue(s)
- closes #1767

### CODE REVIEW

File: python/src/lib.rs [hunk 1]
```diff
@@ -1098,6 +1100,13 @@ fn save_mode_from_str(value: &str) -> PyResult<SaveMode> {
     }
 }
 
+fn partition_strategy_from_str(value: &str) -> PyResult<PartitionStrategy> {
+    match value {
+        "hive" => Ok(PartitionStrategy::Hive),
+        _ => Err(PyValueError::new_err("Invalid partition strategy provided")),
+    }
+}
```
Comment (roeap) [inline python/src/lib.rs:1108]: Would it make sense to have this as a `FromStr` implementation directly on the `PartitionStrategy` enum?

**Review by roeap — COMMENTED** [review]
Thanks for taking care of this!

Left one small comment to look at.

File: python/src/lib.rs (hunk 1 above)
Comment (ion-elgreco) [reply python/src/lib.rs:1108]: Yes, definitely should be a small change. 

Then I guess we can also do this on the savemode enum

Comment (ion-elgreco) [reply python/src/lib.rs]: @roeap I have added it on SaveMode and PartitionStategy

**Review by roeap — DISMISSED** [review]
Looking good! Left two more minor suggestions, that you may choose to include or ignore 😄.

**Comment by ion-elgreco** [comment]
@roeap yeah good one, making input case insensitive!

Btw your approval got dismissed now by committing :😁

**Review by roeap — APPROVED** [review]

**Pull request merged**